pypeit/core/flat.py

```diff
@@ -104,19 +106,18 @@ def fit_flat(flat, tilts_dict, thismask_in, slit_left_in, slit_righ_in, inmask =
     tilts_dict: dict
           Dictionary containing wavelength tilts image and other information indicating how wavelengths move across the slit
 
-    thismask_in:  boolean ndarray, shape (nspec, nspat)
-        Boolean mask image specifying the pixels which lie on the slit/order according to the initial slit/order bounadries.
-        The convention is: True = on the slit/order, False  = off the slit/order
-
-    slit_left_in:  float ndarray, shape  (nspec, 1) or (nspec)
-        Left boundary of slit/order to be extracted (given as floating pt pixels).
-
-    slit_righ_in:  float ndarray, shape  (nspec, 1) or (nspec)
-        Right boundary of slit/order to be extracted (given as floating pt pixels).
-
+    tslits_dict: dict
+          Dictionary with information on the slit boundaries
+    slit: int
+          Slit currently being considered
 
     Optional Parameters
     -------------------
+    slitmask_func: function, default pypeit.spectrographs.spectrograh.Spectrograph.slitmask
```

Comment:
Hi @profxj, 

In my new branch, I've changed this function to take spectrograph as an argument as in the other classes as you suggested. This code only needs one function from the spectrograph class, which is why I wrote it initially this way, but I agree that this is less elegant. 

There is no way to just pass in the slit mask here, because this code tweaks the slits and as such needs to be able to generate the slit mask again. We have violated this rule about not passing objects in other cases, namely in all the extraction and object finding routines which take objects as arguments. I think our philosophy should be avoid it at all costs, except in cases where it makes the code much less tractable. 

Joe